__test__/chat.test.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const chat = require('../chat/chat.js'),
+  io = require('socket.io-client'),
+  io_chat = require('socket.io').listen(3000);
+
+const readline = require('readline');
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+jest.setTimeout(50000);
+  
+
+
+describe('Suite of unit tests', function() {
+
+  let socket;
+  let answer = 'test2';
+
+  beforeEach(function(done) {
+    // Setup
+    socket = io.connect('http://localhost:3000', {
+      'reconnection delay' : 0
+      , 'reopen delay' : 0
+      , 'force new connection' : true,
+    });
+    //socket.on('connect', function() {
+    socket.on('connect', ()=> {
+      console.log(`${answer} joined the chat!`);
+      done();
+    });
+    socket.on('disconnect', function() {
+      console.log(`User left the chat`);
+    });
+  });
+
+  afterEach(function(done) {
+    // Cleanup
+    if(socket.connected) {
+      console.log('disconnecting...');
+      socket.disconnect();
+    } else {
+      // There will not be a connection unless you have done() in beforeEach, socket.on('connect'...)
+      console.log('no connection to break...');
+    }
+    done();
+  });
+
+  describe('Message Events', ()=> {
+  
+    it('Clients should receive a message when the `message` event is emited.', function(done) {
+      let username = 'Test2';
+      socket.emit('username', {username: username});
+      socket.on('line', function(message) {
+        console.log(`${username} joind the chat!`);
+        expect(message).toEqual(`${username} joind the chat!`);
```

Comment:
typo - should be 'joined', not 'joind'

Reply:
Looks like your tests are require server to be up in order for them to pass. We cannot have server up while testing our functionality. We only can mock server. There's NO WAY tests that require running LOCAL server will pass in Travis.